[vinceg4] ip

vinceg4 opened the pull request:
# Bobby

> “Every minute you spend in planning saves 10 minutes in execution; this gives you a 1,000 percent return on energy.” – Brian Tracy ([source](https://www.projectmanager.com/blog/planning-quotes))

Bobby is your **personal butler** 🤵🏻. He helps you to remember the numerous tasks you have and ensures that you are always on track with your tasks!

## Installing Bobby
To get your personal butler:

1. Download the jar file [here](https://github.com/vinceg4/ip/releases/tag/A-Jar)
2. Run the jar file
3. Let your personal butler serve you!

## Features

- Task management
- Save and load your tasks
- `find` matching tasks
- It is **FREE!**

## Upcoming Features

- [x] Add deadlines and events 
- [ ] Add GUI
- [ ] Enhance task addition features

Below is the `main` method for Java users who are interested!
```java
    public static void main(String[] args) {
        Bobby bobby = new Bobby(filePath);
        bobby.run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

vinceg4 committed:
Rename main file

vinceg4 committed:
Add horizontalLine method

vinceg4 committed:
Add greeting

vinceg4 committed:
Add bye() method

vinceg4 committed:
Add greet() method

vinceg4 committed:
Add repeat method

vinceg4 committed:
Import Scanner util

vinceg4 committed:
Add support for echo

vinceg4 committed:
Add storing feature

vinceg4 committed:
Add reply feature after adding task

vinceg4 committed:
Add support for listing out tasks

vinceg4 committed:
Add Task class

vinceg4 committed:
Implement Task class in Bobby class

vinceg4 committed:
Add ability to mark task as done or undone

vinceg4 committed:
Modify Bobby to ignore case of inputs

vinceg4 committed:
Add Todo, Deadline and Event class

vinceg4 committed:
Modify if-else statements to switch

vinceg4 committed:
Modify toString() methods and Event class

vinceg4 committed:
Modify toString() methods

vinceg4 committed:
Add support for Todo, Deadline and Event

vinceg4 committed:
Modify reply after task is added

vinceg4 committed:
Pass A-TextUiTesting

vinceg4 committed:
Add Exception classes for Bobby

vinceg4 committed:
Move creating task functionality from Bobby to the respective Task types classes and implement exception handling.

vinceg4 committed:
Add delete method and functionality

vinceg4 committed:
Add File support for Bobby

vinceg4 committed:
Add support for loading data when Bobby is launched

vinceg4 committed:
Add saving feature to file after any changes to the list of tasks

vinceg4 committed:
Merge branch 'branch-Level-7'

vinceg4 committed:
Ignore data.txt file

vinceg4 committed:
Modify Deadline tasks to work with LocalDate instead of String

vinceg4 committed:
Merge branch 'branch-Level-8'

vinceg4 committed:
Add Ui class

vinceg4 committed:
Add Commit class

vinceg4 committed:
Add UnknownCommandException class

vinceg4 committed:
Add Parser class

vinceg4 committed:
Fix bug in Parser class

vinceg4 committed:
Add TaskList class

vinceg4 committed:
Modify attributes of TaskList

vinceg4 committed:
Add Storage class

vinceg4 committed:
Modify Task and Tasklist class

vinceg4 committed:
Add run method to Ui class

vinceg4 committed:
Modify Command class

vinceg4 committed:
Modify Bobby to work in tandem with the other classes

vinceg4 committed:
Merge branch 'branch-A-MoreOOP'

vinceg4 committed:
Divide classes into packages

vinceg4 committed:
Merge branch 'branch-A-Packages'

vinceg4 committed:
Merge branch 'add-gradle-support'

vinceg4 committed:
Modify build.gradle

vinceg4 committed:
Modify Todo class

vinceg4 committed:
Add Junit tests for Todo, Deadline and Event classes

vinceg4 committed:
Merge branch 'branch-A-JUnit'

vinceg4 committed:
Modify build.gradle shadowJar

vinceg4 committed:
Add JavaDoc comments

vinceg4 committed:
Add JavaDoc comments for Test files and fix the naming of test methods in EventTest and DeadlineTest

vinceg4 committed:
Tweak the code to comply with a coding standard

vinceg4 committed:
Add Find functionality

vinceg4 committed:
Merge branch 'branch-Level-9'

vinceg4 committed:
Merge branch 'branch-A-CodingStandard'

vinceg4 committed:
Resolve merging conflicts

vinceg4 committed:
Add JavaDoc comments for Find functionality and remove unused method from TaskList

vinceg4 committed:
Classify classes into different packages

vinceg4 committed:
Update build.gradle for JavaFX

vinceg4 committed:
Modify build.gradle

vinceg4 committed:
Add images for Bobby GUI

vinceg4 committed:
Add support for JavaFX

vinceg4 committed:
Add support for GUI

vinceg4 committed:
Fix bugs for certain commands

vinceg4 committed:
Merge branch 'branch-Level-10'

vinceg4 committed:
Add checkstyle

vinceg4 committed:
Merge branch 'branch-A-CheckStyle'

KrashKart commented on src/main/java/bobby/utils/Ui.java:
```diff
@@ -0,0 +1,109 @@
+     * Prints the greeting message of Bobby.
+     */
+    public String showGreeting() {
+        String output = "";
```
I like your use of camelCase for all your methods!

KrashKart commented on src/main/java/bobby/Bobby.java:
```diff
@@ -0,0 +1,174 @@
+
+    /**
+     * The method that is run when Bobby is launched
+     */
```
perhaps you could add a full stop after this Javadoc statement?

KrashKart commented on src/main/java/bobby/exception/UnknownCommandException.java:
```diff
@@ -0,0 +1,14 @@
+ * UnknownCommandException is an exception thrown when an unknown task type is provided.
+ */
+public class UnknownCommandException extends BobbyException {
+
```
Nice use of PascalCase for all your class names!

KrashKart commented on src/main/java/bobby/gui/DialogBox.java:
```diff
@@ -0,0 +1,60 @@
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
```
Nice proper ordering of imports!

KrashKart reviewed (commented):
LGTM! Hi Vince 👋🏻

jolynleow2301 commented on src/main/java/bobby/utils/Command.java:
```diff
@@ -0,0 +1,16 @@
+package bobby.utils;
+
+public enum Command {
+    /**
```
Header comment should be made before the start of enum Command

jolynleow2301 commented on src/main/java/bobby/utils/Storage.java:
```diff
@@ -0,0 +1,103 @@
+     * If there was no file containing the tasks, a new file will be created and empty ArrayList would be returned.
+     *
+     * @return ArrayList of Task
+     */
```
Good header comments written :)

jolynleow2301 commented on src/main/java/bobby/utils/Ui.java:
```diff
@@ -0,0 +1,109 @@
+/**
+ * The Ui class deals with any user interaction.
+ */
+public class Ui {
```
Good adherence to Coding Standards for all your methods

jolynleow2301 commented on src/main/java/bobby/utils/Parser.java:
```diff
@@ -0,0 +1,66 @@
+        String[] splitInput = userInput.split(" ");
+        if (splitInput[0].equalsIgnoreCase("bye")) {
+            return Command.BYE;
+        }
```
can consider using "else if"

jolynleow2301 commented on src/main/java/bobby/tasks/Todo.java:
```diff
@@ -0,0 +1,57 @@
+        return new Todo(todoDescription);
+    }
+    /**
+     * @inheritDoc
```
Good that you've included header comments for all your methods :)

vinceg4 committed:
Checkstyle: Modify Javadoc

Modify Javadoc comments to conform to Checkstyle requirements.

vinceg4 committed:
Add Assertions

No assertions used in the current code.

Assumptions about the program state cannot be checked, leading to
higher chance of runtime errors.

Let's add assertions to check that a file has been created in the
returnTaskList() function present in the Storage class.

vinceg4 committed:
CodeQuality: Improve code quality

There were several issues regarding the code which did not comply
with the given code convention.

A clearer and more concise documentation serve a better role in
understanding and debugging the code in the future.

Let's
* add in missing Javadoc comments
* use clearer variable names
* add in default branch for case statements
* rectify wrong import orders
* remove unused code

vinceg4 committed:
Merge pull request #1 from vinceg4/branch-A-Assertions

Add Assertions

vinceg4 committed:
Merge branch 'master' into branch-A-CodeQuality

vinceg4 committed:
Merge pull request #2 from vinceg4/branch-A-CodeQuality

CodeQuality: Improve code quality

vinceg4 committed:
Add support for archiving tasks.

There was no functionality to archive tasks.

Completed tasks may accumulate and result in a very long list
of tasks.

Let's provide a way for users to archive their tasks and view
a list of their archived tasks.

vinceg4 committed:
Merge branch 'branch-C-Archive'

vinceg4 committed:
Improve GUI for Bobby

vinceg4 committed:
Merge branch 'branch-A-BetterGui'

vinceg4 committed:
Fix Javadocs comments

vinceg4 committed:
Add Javadocs comment

vinceg4 committed:
Fix checkstyle violations

vinceg4 committed:
Fix bugs regarding deletion of tasks

vinceg4 committed:
Add Ui.png

vinceg4 committed:
Add user guide in README.md

vinceg4 committed:
Merge branch 'branch-A-UserGuide'

vinceg4 committed:
Update README.md

vinceg4 committed:
Update README.md

vinceg4 committed:
Fix checkstyle violations

vinceg4 committed:
Fix bugs in error handling

vinceg4 committed:
Update README.md
